src/components/CardProfile.vue

```diff
@@ -0,0 +1,218 @@
+<template>
+	<v-card class="card-profile" max-width="800" height="380px">
+		<div class="content-card">
+			<v-avatar color="black" size="70" class="mb-3 icon-profile">
+				<img :src="avatarUrl" alt="User Profile" width="100%" height="100%" />
+			</v-avatar>
+
+			<span class="username">{{ user.username }}</span>
+			<span class="email">{{ user.email }}</span>
+			<input type="file" class="input-file" @change="uploadImage" />
+		</div>
+		<v-card-actions class="button">
+			<v-btn variant="tonal" @click="dialog = true">Editar</v-btn>
+			<v-btn variant="tonal" @click="goToHome">Voltar</v-btn>
+		</v-card-actions>
+	</v-card>
+
+	<!-- Atualizar Cadastro como Popup -->
+	<v-dialog v-model="dialog" max-width="400px" persistent>
+		<v-card>
+			<v-card-title class="text-center">
+				<span class="title-edit">Atualizar Cadastro</span>
+			</v-card-title>
+
+			<v-form ref="form" validate-on="input">
+				<v-text-field
+					name="username"
+					label="Username"
+					variant="filled"
+					:rules="usernameRules"
+				></v-text-field>
+				<v-text-field
+					name="password"
+					v-model="password"
+					:type="showPassword ? 'text' : 'password'"
+					color="primary"
+					label="Senha"
+					placeholder="Enter your password"
+					variant="filled"
+					:append-icon="showPassword ? 'mdi-eye-off' : 'mdi-eye'"
+					:rules="passwordRules"
+					@click:append="toggleShowPassword"
+				></v-text-field>
+			</v-form>
+
+			<v-divider></v-divider>
+
+			<v-card-actions>
+				<v-spacer></v-spacer>
+				<v-btn color="success" @click="dialog = false">
+					<span style="color: #ffd200">Cancelar </span>
+				</v-btn>
+				<v-btn color="success" @click="updatePassword">
+					<span style="color: #ffd200">Confirmar </span>
+				</v-btn>
+			</v-card-actions>
+		</v-card>
+	</v-dialog>
+</template>
+
+<script>
+import { userStore } from "@/store/userStore";
+import { mapState } from "pinia";
+import { uuid } from "vue-uuid";
+import { supabase } from "@/config/supabase";
+
+export default {
+	data() {
+		return {
+			dialog: false,
+			first: "",
+			email: "",
+			password: "",
+			avatarUrl: "", // Inicializa o avatarUrl
+			showPassword: false,
+			usernameRules: [
+				(v) =>
+					(v && v.length >= 3) || "O username deve contar 3 ou mais caracteres",
+			],
+			emailRules: [(v) => this.validateEmail(v) || "E-mail deve ser válido"],
+			passwordRules: [
+				(v) => v.length >= 6 || "A senha deve ter pelo menos 6 caracteres",
+				(v) =>
+					this.validatePassword(v) ||
+					"A senha deve conter um caractere especial",
+			],
+		};
+	},
+	async created() {
+		const { data: user } = await supabase.auth.getUser();
+
+		// Tenta carregar a URL do avatar do localStorage
+		const savedUser = JSON.parse(localStorage.getItem("user"));
+		if (savedUser && savedUser.avatarUrl) {
+			this.avatarUrl = savedUser.avatarUrl; // Carrega do localStorage
+		} else if (user && user.user_metadata) {
```

Comment:
Não estou lembrado se isso é suportado pelo vue.js na versão que estamos utilizando nesse projeto, mas existe uma short-syntax para ter apenas uma condicional ao invés de duas utilizando um optional chaining `?.`:
```suggestion
		if (savedUser?.avatarUrl) {
			this.avatarUrl = savedUser.avatarUrl; // Carrega do localStorage
		} else if (user?.user_metadata) {
```

Leia mais sobre esse tópico, é uma feature do es5/es6 muito utilizada, existe uma grande chance de você encontrar esse tipo de código em algum projeto que você trabalhará: [Optional Chaining](https://medium.com/@guigaoliveira_/entendendo-o-optional-chaining-no-javascript-964ca6928598)

Reply:
![image](https://github.com/user-attachments/assets/63ab2cd2-4c41-4235-8c73-65fec7929919)